OpenRA.Mods.Common/Irc.cs

```diff
@@ -0,0 +1,21 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Threading;
+using Meebey.SmartIrc4net;
+
+namespace OpenRA.Mods.Common
+{
+	public static class Irc
+	{
+		public static IrcClient Client;
```

Comment:
We'll need to access this in a thread-safe way - although the class is itself thread-safe because we sometimes do several operations in a row on it, it's state may have changed. For example, we may do a null check, find it not-null. Then it might disconnect under our feet, and when we access some property on it we find it's suddently become null again.

A static object to use in a bunch of lock statements would help with that. We probably want to make both ```Client``` and this lock object private to the ```IrcLogic``` to avoid exposing them elsewhere (somewhat nasty as that is).